digiroad2-viite/src/main/scala/fi/liikennevirasto/viite/RoadAddressService.scala

```diff
@@ -246,19 +247,35 @@ class RoadAddressService(roadLinkService: RoadLinkService, roadwayDAO: RoadwayDA
     var resultSeq: Seq[Map[String, Seq[Any]]] = Seq.empty[Map[String, Seq[Any]]]
     if (searchType == "road") {
       val nums = parsedInput.head._2
+      val roadNumberOrId = nums(0)
       if (nums.size == 3) {
-        searchResult = getRoadAddress(nums(0), nums(1), nums(2), None).sortBy(address => (address.roadPartNumber, address.startAddrMValue))
-        resultSeq = collectResult("road", searchResult, resultSeq)
+        val roadPart = nums(1)
+        val addressM = nums(2)
+        val ralOption = getRoadAddressLink(roadNumberOrId, roadPart, addressM)
+        if (ralOption.isDefined) {
+          val ral = ralOption.get
+          val points = ral.geometry
+          val geometryMeasure = ((addressM - ral.startAddressM.toDouble) / (ral.endAddressM.toDouble - ral.startAddressM)) * (ral.endMValue - ral.startMValue)
+          val mValue: Double = ral.sideCode match {
+            case AgainstDigitizing => (ral.endMValue - geometryMeasure)
+            case _ => geometryMeasure
+          }
+          val point = GeometryUtils.calculatePointFromLinearReference(points,mValue)
+          resultSeq = collectResult("roadM", Seq(point), resultSeq)
+        } else {
+          logger.info(s"""Search found nothing with: $searchString""")
+        }
```

Comment:
Can be done in a more "Scala" way, for example: 

`        ralOption.map { ral =>
          val points = ral.geometry
          val geometryMeasure = ((addressM - ral.startAddressM.toDouble) / (ral.endAddressM.toDouble - ral.startAddressM)) * (ral.endMValue - ral.startMValue)
          val mValue: Double = ral.sideCode match {
            case AgainstDigitizing => (ral.endMValue - geometryMeasure)
            case _ => geometryMeasure
          }
          val point = GeometryUtils.calculatePointFromLinearReference(points, mValue)
          resultSeq = collectResult("roadM", Seq(point), resultSeq)
        }.getOrElse(logger.info(s"""Search found nothing with: $searchString"""))`

Reply:
badly formatted, here it's better: https://gist.github.com/oskarmalmstrom/cd1c8446199fc7cc04dc57efb828ceb4

Reply:
fixed